phpBB/phpbb/controller/helper.php

```diff
@@ -142,6 +142,12 @@ public function route($route, array $params = array(), $is_amp = true, $session_
 
 		$base_url = $context->getBaseUrl();
 
+		// Regarding PHPBB3-13493 - if we request url ending on / there is no $page_name
+		if (!strpos($base_url, $page_name))
```

Comment:
$page_name is never empty. It is taken from https://github.com/satanasov/phpbb/blob/ticket/13493/phpBB/phpbb/controller/helper.php#L141

This means that it will always have  $page_name. But you are right for the check.